src/bp/ui-studio/src/web/views/OneFlow/sidePanel/Library/index.tsx

```diff
@@ -170,30 +171,17 @@ const Library: FC<Props> = props => {
     }
   }
 
-  const nextFlowName = (topic: string, originalName: string): string => {
-    let name = undefined
-    let fullName = undefined
-    let index = 0
-    do {
-      name = `${originalName}${index ? `-${index}` : ''}`
-      fullName = buildFlowName({ topic, workflow: name }, true).workflowPath
-      index++
-    } while (props.flows.find(f => f.name === fullName))
-
-    return fullName
-  }
-
```

Comment:
Instead of moving this method and passing it as a property, you can extract it somewhere and add a third argument to provide the list of flows. This way you can use it here and in the other place, while avoiding a new property